projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_base.py

```diff
@@ -512,12 +526,8 @@ def _payload_poster_thread(self):
                         metadata=ingestion_metadata,
                         exception=exception,
                     )
-            except VdkConfigurationError:
-                self._plugin_errors[VdkConfigurationError].increment()
-            except UserCodeError:
-                self._plugin_errors[UserCodeError].increment()
-            except Exception:
-                self._plugin_errors[PlatformServiceError].increment()
+            except Exception as e:
+                self._plugin_errors[type(e)].increment()
```

Comment:
Not sure this will work as intended. We do this later on

```
    def __handle_results(self):
        if self._plugin_errors.get(UserCodeError, AtomicCounter(0)).value > 0:
            errors.report_and_throw(
                UserCodeError(
                    "Failed to post all data for ingestion successfully.",
                    "Some data will not be ingested.",
                    "Ensure data you are sending is aligned with the requirements",
                    "User error occurred. See warning logs for more details. ",
                )
            )
        if self._plugin_errors.get(VdkConfigurationError, AtomicCounter(0)).value > 0:
            errors.report_and_throw(
                VdkConfigurationError(
                    "Failed to post all data for ingestion successfully.",
                    "Some data will not be ingested.",
                    "Ensure job is properly configured. "
                    "For example make sure that target and method specified are correct",
                )
            )
        if (
            self._plugin_errors.get(PlatformServiceError, AtomicCounter(0)).value > 0
            or self._fail_count.value > 0
        ):
            errors.report_and_throw(
                PlatformServiceError(
                    "Failed to post all data for ingestion successfully.",
                    "Some data will not be ingested.",
                    "There has been temporary failure. ",
                    "You can retry the data job again. ",
                    "If error persist inspect logs and check ingest plugin documentation.",
                )
            )
```

It might be a better idea to map by exception type, collect the exceptions for the appropriate type and decide what to do with them.